tests/_utils-tests/view.js

```diff
@@ -0,0 +1,423 @@
+			selection.addRange( range );
+			expect( stringify( p, selection ) ).to.equal( '<p><b>f{ooba}r</b><b>bazqux</b></p>' );
+		} );
+
+		it( 'should write collapsed selection ranges inside texts', () => {
+			const text = new Text( 'foobar' );
+			const p = new Element( 'p', null, text );
+			const range = Range.createFromParentsAndOffsets( text, 0, text, 0 );
+			const selection = new Selection();
+			selection.addRange( range );
+			expect( stringify( p, selection ) ).to.equal( '<p>{}foobar</p>' );
+		} );
+
+		it( 'should write ranges that start inside text end ends between elements', () => {
+			const text1 = new Text( 'foobar' );
+			const text2 = new Text( 'bazqux' );
+			const b1 = new Element( 'b', null, text1 );
+			const b2 = new Element( 'b', null, text2 );
+			const p = new Element( 'p', null, [ b1, b2 ] );
+			const range = Range.createFromParentsAndOffsets( p, 0, text2, 5 );
+			const selection = new Selection();
+			selection.addRange( range );
+			expect( stringify( p, selection ) ).to.equal( '<p>[<b>foobar</b><b>bazqu}x</b></p>' );
+		} );
+
+		it( 'should write elements types as namespaces when needed', () => {
+			const text = new Text( 'foobar' );
+			const b = new AttributeElement( 'b', null, text );
+			const p = new ContainerElement( 'p', null, b );
+
+			expect( stringify( p, null, { showType: true } ) )
+				.to.equal( '<container:p><attribute:b>foobar</attribute:b></container:p>' );
+		} );
+
+		it( 'should not write element type when type is not specified', () => {
+			const p = new Element( 'p' );
+			expect( stringify( p, null, { showType: true } ) ).to.equal( '<p></p>' );
+		} );
+
+		it( 'should write elements priorities when needed', () => {
+			const text = new Text( 'foobar' );
+			const b = new AttributeElement( 'b', null, text );
+			const p = new ContainerElement( 'p', null, b );
+
+			expect( stringify( p, null, { showPriority: true } ) )
+				.to.equal( '<p><b:10>foobar</b:10></p>' );
+		} );
+
+		it( 'should parse DocumentFragment as root', () => {
+			const text1 = new Text( 'foobar' );
+			const text2 = new Text( 'bazqux' );
+			const b1 = new Element( 'b', null, text1 );
+			const b2 = new Element( 'b', null, text2 );
+			const fragment = new DocumentFragment( [ b1, b2 ] );
+			expect( stringify( fragment, null ) ).to.equal( '<b>foobar</b><b>bazqux</b>' );
+		} );
+
+		it( 'should not write ranges outside elements', () => {
+			const text = new Text( 'foobar' );
+			const b = new Element( 'b', null, text );
+			const p = new Element( 'p', null, b );
+			const range1 = Range.createFromParentsAndOffsets( p, 0, p, 5 );
+			const range2 = Range.createFromParentsAndOffsets( p, -1, p, 1 );
+			const range3 = Range.createFromParentsAndOffsets( text, 0, text, 7 );
+			const range4 = Range.createFromParentsAndOffsets( text, -1, text, 2 );
+			const range5 = Range.createFromParentsAndOffsets( text, 6, text, 8 );
+			const selection = new Selection();
+			selection.addRange( range1 );
```

Comment:
I'm missing vertical spacing in this code. It's made of logical parts - preparation, case1, case2, case3, etc. Blank lines would make it less overwhelming :D.